src/passes/RemoveUnusedBrs.cpp

```diff
@@ -556,11 +562,169 @@ struct RemoveUnusedBrs : public WalkerPass<PostWalker<RemoveUnusedBrs>> {
           }
         }
       }
+
+      // (br_if)+ => br_table
+      // we look for the specific pattern of
+      //  (br_if ..target1..
+      //    (i32.eq
+      //      (..input..)
+      //      (i32.const ..value1..)
+      //    )
+      //  )
+      //  (br_if ..target2..
+      //    (i32.eq
+      //      (..input..)
+      //      (i32.const ..value2..)
+      //    )
+      //  )
+      // TODO: consider also looking at <= etc. and not just eq
+      void tablify(Block* block) {
+        auto &list = block->list;
+        if (list.size() <= 1) return;
+
+        // Heuristics. These are slightly inspired by the constants from the asm.js backend.
+
+        // How many br_ifs we need to see to consider doing this
+        const uint32_t MIN_NUM = 3;
+        // How much of a range of values is definitely too big
+        const uint32_t MAX_RANGE = 1024;
+        // Multiplied by the number of br_ifs, then compared to the range. When
+        // this is high, we allow larger ranges.
+        const uint32_t NUM_TO_RANGE_FACTOR = 3;
+
+        // check if the input is a proper br_if on an i32.eq of a condition value to a const,
+        // and the const is in the proper range, [0-int32_max), to avoid overflow concerns.
+        // returns the br_if if so, or nullptr otherwise
+        auto getProperBrIf = [](Expression* curr) -> Break*{
+          auto* br = curr->dynCast<Break>();
+          if (!br) return nullptr;
+          if (!br->condition || br->value) return nullptr;
+          if (br->type != none) return nullptr; // no value, so can be unreachable or none. ignore unreachable ones, dce will clean it up
+          auto* binary = br->condition->dynCast<Binary>();
+          if (!binary) return nullptr;
+          if (binary->op != EqInt32) return nullptr;
+          auto* c = binary->right->dynCast<Const>();
```

Comment:
Might it make sense to test for the inverse as well, i.e., the constant in `left` and the condition in `right`?

Reply:
The optimize-instructions pass will canonicalize things so a comparison of a const to anything else leaves the const on the right. And it makes sense to run that pass before this one anyhow, I think?

Reply:
I see, didn't know that. How little I know after all :)

Reply:
Heh, well, it's an internal optimizer detail. Should probably be documented somehow.